src/huggingface_hub/cli/repo.py

```diff
@@ -21,227 +21,239 @@
-        tag_list_parser.add_argument(
-            "repo_id", type=str, help="The ID of the repo to list tags for (e.g. `username/repo-name`)."
-        )
-        tag_list_parser.add_argument(
-            "--token", type=str, help="A User Access Token generated from https://huggingface.co/settings/tokens."
-        )
-        tag_list_parser.add_argument(
-            "--repo-type",
-            choices=["model", "dataset", "space"],
-            default="model",
+        ),
+    ] = "model",
+) -> None:
+    repo_type = validate_repo_type(repo_type) or "model"
+    api = HfApi(token=token)
+    print(f"You are about to create tag {ANSI.bold(tag)} on {repo_type} {ANSI.bold(repo_id)}")
+    try:
+        api.create_tag(repo_id=repo_id, tag=tag, tag_message=message, revision=revision, repo_type=repo_type)
+    except RepositoryNotFoundError:
+        print(f"{repo_type.capitalize()} {ANSI.bold(repo_id)} not found.")
+        raise typer.Exit(code=1)
+    except RevisionNotFoundError:
+        print(f"Revision {ANSI.bold(str(revision))} not found.")
+        raise typer.Exit(code=1)
+    except HfHubHTTPError as e:
+        if e.response.status_code == 409:
+            print(f"Tag {ANSI.bold(tag)} already exists on {ANSI.bold(repo_id)}")
+            raise typer.Exit(code=1)
+        raise e
+    print(f"Tag {ANSI.bold(tag)} created on {ANSI.bold(repo_id)}")
+
+
+@tag_app.command("list", help="List tags for a repo.")
+def tag_list(
+    repo_id: Annotated[
+        str,
+        typer.Argument(
+            ...,
+            help="The ID of the repo to list tags for (e.g. `username/repo-name`",
+        ),
+    ],
+    token: Annotated[
+        Optional[str],
+        typer.Option(
+            help="A User Access Token generated from https://huggingface.co/settings/tokens.",
+        ),
+    ] = None,
+    repo_type: Annotated[
+        Optional[str],
+        typer.Option(
             help="Set the type of repository (model, dataset, or space).",
-        )
-        tag_list_parser.set_defaults(func=lambda args: RepoTagListCommand(args))
-
-        # tag delete
-        tag_delete_parser = tag_subparsers.add_parser("delete", help="Delete a tag from a repo.")
-        tag_delete_parser.add_argument(
-            "repo_id", type=str, help="The ID of the repo to delete the tag from (e.g. `username/repo-name`)."
-        )
-        tag_delete_parser.add_argument("tag", type=str, help="The name of the tag to delete.")
-        tag_delete_parser.add_argument("-y", "--yes", action="store_true", help="Answer Yes to prompts automatically.")
-        tag_delete_parser.add_argument(
-            "--token", type=str, help="A User Access Token generated from https://huggingface.co/settings/tokens."
-        )
-        tag_delete_parser.add_argument(
-            "--repo-type",
-            choices=["model", "dataset", "space"],
-            default="model",
+        ),
+    ] = "model",
```

Comment:
I feel that since these definitions are always the same we could centralize the main arguments definition (like repo_id, repo_type, token, revision, etc.) and reuse them in all commands. It would reduce LoC to maintain + make sure that the wording is always consistent. Here for example if could be reduced to

```py
@tag_app.command("list", help="List tags for a repo.")
def tag_list(
    repo_id: RepoIdArg,
    token: TokenOpt = None,
    repo_type: RepoTypeOpt = RepoType.model,
) -> None:
```

We can argue it makes the code slightly less self-explicit but I still believe it's best for consistency.